pkg/operator/controller/controller_dns_configmap.go

```diff
@@ -63,3 +74,12 @@ func desiredDNSConfigMap(dns *operatorv1.DNS, clusterDomain string, daemonsetRef
 	}
 	return cm
 }
+
+func corefileChanged(current, expected *corev1.ConfigMap) (bool, *corev1.ConfigMap) {
+	if cmp.Equal(current.Data, expected.Data, cmpopts.EquateEmpty()) {
```

Comment:
This is fine, but out of curiosity, is there a reason not to use `reflect.DeepEqual` here?

Reply:
I want to say it was actually you who started proliferating the use of [go-cmp](https://github.com/google/go-cmp) into the operators 😁 — for the most part I try to just use it by default now for consistency and flexibility

Reply:
Ravi introduced the first use of go-cmp, and mostly we use it where `reflect.DeepEqual` is not flexible enough, but yeah, we're now using go-cmp more than `reflect.DeepEqual`, so it is reasonable to choose go-cmp over `reflect.DeepEqual` by default.